### PR TITLE
feat: add unmarshal utility functions for primitive types

### PR DESCRIPTION
**Pull request by padamstx**
Fixes: https://github.ibm.com/arf/planning-sdk-squad/issues/1148

This PR adds some utility methods for unmarshalling primitive types.  These additions are part of the renovation of the Go generator to change the way in which we de-serialize operation response bodies, which will also allow us to more easily support discriminators.

Edit: as part of this PR, I also got rid of the "v3" version directory (everything that was in "v3" was just moved up to the root of the project).   I had initially created the "v3" directory when I created the v3.0.0 major version of the core because my understanding was that the version directory was a firm requirement when creating a new major version of a go package (module).   It turns out that it's not quite as firm as I was led to believe :)   The actual requirement is that there must be a go.mod file in the project root directory or one of its subdirectories that contains `module github.com/IBM/go-sdk-core/v3` at the top.   It doesn't have to be located in the "v3" directory.  Also, this change will allow us to continue to use semantic-release.

### CODE REVIEW

**Comment by padamstx** [comment]
@mkistler This PR is a prereq to the go discriminator work in the generator.   I wanted to make sure you had something to do on the plane ride back home tomorrow :)

File: core/unmarshal_test.go [hunk 1]
```diff
@@ -0,0 +1,639 @@
...
+	t.Logf("Expected error: %s\n", err.Error())
+
+	slice, err = UnmarshalDateSlice(testMap, "XXX")
+	assert.Nil(t, err)
+	assert.Nil(t, slice)
+}
+
+func TestUnmarshalDateTime(t *testing.T) {
+	datetime1 := "1970-01-01T01:02:03"
+	datetime2 := "2019-12-23T23:59:59"
```
Comment (mkistler) [inline core/unmarshal_test.go]: The [API Handbook](https://test.cloud.ibm.com/docs/api-handbook?topic=api-handbook-types#date-time)  requires date-time values to be in one of two specific formats:
- `YYYY-MM-DDTHH:mm:ssZ`
- `YYYY-MM-DDTHH:mm:ss.sssZ`

You should change this test to verify correct unmarshalling of both of these date formats.

File: core/unmarshal.go [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+
+// UnmarshalInt64 retrieves the specified property from the map and returns is as an int64
+// or nil if the property wasn't found in the map.
+func UnmarshalInt64(m map[string]interface{}, propertyName string) (result *int64, err error) {
+	var v interface{}
+	v, foundIt := m[propertyName]
+	if foundIt {
+		f, ok := v.(float64)
+		if ok {
+			result = Int64Ptr(int64(f))
```
[the remark this answers is not in the document]
Comment (padamstx) [reply core/unmarshal.go]: Yes, there is a good reason :)   (updated comments in code)

**Comment by padamstx** [comment]
@mkistler I've addressed your review comments.  Ready for another look.

**Pull request merged**